src/solve.jl

```diff
@@ -156,6 +156,13 @@ function DiffEqBase.__init(
 
   callbacks_internal = CallbackSet(callback,prob.callback)
 
+  max_len = DiffEqBase.max_vector_callback_length(callbacks_internal)
+  if max_len !== -1
```

Comment:
not a huge deal, but breaks inference. Can we instead just search for the existence of a VectorContinuousCallback?

Reply:
how does it break inference, return type is always Int64? I would need the max length out of all VectorContinuousCallback to make the CallbackCache.

Reply:
`max_len` cannot be inferred since it's a value, and then which of the branches is taken isn't inferred.

Reply:
how do you propose to do this? I think `Val{}` might not be of use as well

Reply:
Look for VectorContinuousCallback. That is type information.